.ado/jobs/jschecks.yml

```
@@ -17,26 +17,11 @@ jobs:
     timeoutInMinutes: 30 # how long to run the job before automatically cancelling
     cancelTimeoutInMinutes: 5 # how much time to give 'run always even if cancelled tasks' before killing them
     steps:
-      - checkout: self # self represents the repo where the initial Pipelines YAML file was found
-        clean: true # whether to fetch clean each time
-        lfs: false # whether to download Git-LFS files
-        submodules: false # set to 'true' for a single level of submodules or 'recursive' to get submodules of submodules
-        persistCredentials: false # set to 'true' to leave the OAuth token in the Git config after the initial fetch
+      - checkout: self
+        fetchDepth: 2
```

I would recommend at least a depth of 10. I have seen too many git failures with a low fetch-depth where the sync fails if a commit happens during the sync and it turned out the depth was not enough...

[shallow-fetch docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/pipeline-options-for-git?view=azure-devops&tabs=yaml#shallow-fetch)
[Me sharing a build error where it happened with wrong link :)](https://github.com/MicrosoftDocs/azure-devops-docs/issues/3453)



```suggestion
        fetchDepth: 10
```

---

Pack Desktop has been running with 2 for quite some time without failures, but I can imagine where this would happen.

Cleaning up the other usages in our repo to specify intent of `checkout-shallow` or `checkout-full` to tuck these bits away into a template and encourage new jobs to think about what they need.